src/App.tsx

```diff
@@ -1,27 +1,94 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+import { Loader } from './Loader';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+enum FilterOptions {
+  ALL = 'all-button',
+  FIRST5 = 'first-five-button',
+  RED = 'red-button',
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+type ButtonNames = {
+  [key in FilterOptions]: string;
+};
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+const buttonNames: ButtonNames = {
+  [FilterOptions.ALL]: 'Load all goods',
+  [FilterOptions.FIRST5]: 'Load 5 first goods',
+  [FilterOptions.RED]: 'Load red goods',
+};
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [filterOptions, setFilterOptions] = useState('');
```

Comment:
Consider renaming the filterOptions state variable to something more descriptive, like currentFilter.